WRR-287: Replace CSS variables having R,G,B formats with standard #RGB formats

0x64 opened the pull request:
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
CSS variables for customization of apps on run-time is introduced in 2.1.0 and we used some R,G,B format values for colors that need to be translucent on some UI components. At that time, we had no proper way to combine #RGB format with opacity value in CSS semantics.
As of now, we can use relative RGB semantics and it is a good time to turn non-standard R,G,B formats into standard #RGB formats.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
In this pull request, the followings are done.
* R,G,B format values becomes #RGB values.
* CSS/LESS variables having suffix -rgb are renamed the same without suffix -rgb.
* Some CSS/LESS variables are refined/renamed to distinguish opacity (with suffix -opacity) for CSS opacity property and opacity (with suffix -color-opacity) for colors.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Replaced CSS variables (corresponding LESS variables are also replaced)
--sand-text-color-rgb > --sand-text-color
--sand-shadow-color-rgb > --sand-shadow-color
--sand-component-text-color-rgb > --sand-component-text-color
--sand-component-text-sub-color-rgb > --sand-component-text-sub-color
--sand-focus-bg-color-rgb > --sand-focus-bg-color
--sand-component-focus-text-color-rgb > --sand-component-focus-text-color
--sand-selected-color-rgb > --sand-selected-color
--sand-overlay-bg-color-rgb > --sand-overlay-bg-color
--sand-progress-color-rgb > --sand-progress-color
--sand-progress-bg-color-rgb > --sand-progress-bg-color
--sand-keyguide-bg-color-rgb > --sand-keyguide-bg-color
--sand-alert-overlay-bg-color-rgb > --sand-alert-overlay-bg-color
--sand-alert-overlay-text-color-rgb > --sand-alert-overlay-text-color
--sand-alert-overlay-progress-color-rgb > --sand-alert-overlay-progress-color
--sand-alert-overlay-progress-bg-color-rgb > --sand-alert-overlay-progress-bg-color

Modified CSS variables
--sand-progress-bg-color-opacity is now a number between 0 and 1 instead of a percentage.
--sand-spinner-color is now #hex format instead of {R,G,B} format. (It was a bug that we did not name it --sand-spinner-color-rgb before).
--sand-alert-overlay-progress-bg-color-opacity is now a number between 0 and 1 instead of a percentage.

Renamed CSS variables (corresponding LESS variables are also renamed)
--sand-progress-bg-color-alpha > --sand-progress-bg-color-opacity
--sand-alert-overlay-progress-bg-color-alpha > --sand-alert-overlay-progress-bg-color-opacity

New CSS variables
--sand-keyguide-bg-color-opacity is added to support overriding.
--sand-steps-pageindicator-bg-color as we missed to add this non-common component-specific color.

New LESS variables
@sand-shadow-base-color is added as a base color without opacity.
@sand-keyguide-bg-base-color and @sand-keyguide-bg-color-opacity are added to override easily for skins.
@sand-disabled-focus-content-color-opacity is added to separate from @sand-disabled-focus-content-opacity that is used as an opacity for whole control not only text.

### Links
[//]: # (Related issues, references)
WRR-287

### Comments
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

MikyungKim commented on Alert/Alert.module.less:
```diff
@@ -138,22 +138,22 @@
 			--sand-item-disabled-focus-bg-color: @sand-alert-overlay-item-disabled-focus-bg-color;
 
-			background-color: rgb(@sand-alert-overlay-bg-color-rgb, @sand-alert-overlay-bg-color-opacity);
+			background-color: ~"color(from" @sand-alert-overlay-bg-color ~"srgb r g b /" @sand-alert-overlay-bg-color-alpha ~")";
```
Is there a reason for renaming `-opacity` to `alpha`?

0x64 replied:
Discussing with @MikyungKim, we'd like to change color-alpha to color-opacity.

0x64 replied:
Note: from now, -color-opacity for opacity of a color like text color or background color and -opacity (without color in front of it) for opacity of a component

MikyungKim commented on styles/colors-game.less:
```diff
@@ -30,9 +30,9 @@
-@sand-alert-overlay-text-color-rgb: var(--sand-alert-overlay-text-color-rgb, 255, 255, 255); // #ffffff
-@sand-alert-overlay-bg-color-rgb: var(--sand-alert-overlay-bg-color-rgb, 0, 0, 0); // #000000
-@sand-alert-overlay-bg-color-opacity: 75%;
+@sand-alert-overlay-bg-color: var(--sand-alert-overlay-bg-color, rgb(var(--sand-alert-overlay-bg-color-rgb, 0, 0, 0))); // #000000
```
What is `--sand-alert-overlay-bg-color-rgb`?

0x64 replied:
It is a kind of fallback for old legacy uses. I will update the PR description.

0x64 replied:
Updated

MikyungKim commented on styles/colors.less:
```diff
@@ -19,29 +19,23 @@
-@sand-component-text-sub-color-rgb: var(--sand-component-text-sub-color-rgb, 171, 174, 179); // #abaeb3
-@sand-component-text-sub-color: rgb(@sand-component-text-sub-color-rgb);
+@sand-shadow-color: var(--sand-shadow-color, rgb(var(--sand-shadow-color-rgb, 0, 0, 0))); // #000000
+@sand-shadow-common-color: ~"color(from" @sand-shadow-color ~"srgb r g b / 0.3)";
```
Let's discuss the naming again.

0x64 replied:
Conclusion: @sand-shadow-base-color for @sand-shadow-color, and @sand-shadow-color for @sand-shadow-common-color as the current @sand-shadow-common-color is the basic shadow color.

0x64 replied:
done

MikyungKim commented on styles/colors.less:
```diff
@@ -132,14 +123,15 @@
 @sand-disabled-opacity: 0.28;  // This value brings the standard text-color to the disabled text color specs
 @sand-disabled-bg-opacity: @sand-disabled-opacity;
 @sand-disabled-focus-content-opacity: 0.4;
+@sand-disabled-focus-content-color-alpha: 0.4;
```
Is this necessary?

0x64 replied:
done

MikyungKim commented on styles/colors.less:
```diff
@@ -98,18 +89,18 @@
 @sand-formcheckboxitem-focus-text-color: var(--sand-formcheckboxitem-focus-text-color, @sand-focus-text-color);
 @sand-item-disabled-focus-bg-color: var(--sand-item-disabled-focus-bg-color, #e6e6e6);
-@sand-keyguide-bg-color-rgb: var(--sand-keyguide-bg-color-rgb, 55, 58, 65); // #373a41
+@sand-keyguide-bg-color: var(--sand-keyguide-bg-color, rgb(var(--sand-keyguide-bg-color-rgb, 55, 58, 65))); // #373a41
+@sand-keyguide-bg-color-alpha: var(--sand-keyguide-bg-color-alpha, 0.9);
```
As we discussed let's rename `@sand-keyguide-bg-color-alpha` to `@sand-keyguide-bg-color` with color css function and `@sand-keyguide-bg-color` to `@sand-keyguide-bg-base-color`.

0x64 replied:
I updated the code and it looks good to leave @sand-keyguide-bg-color-opacity since it makes easy to change opacity in high-contrast skin variants just like other components.

0x64 commented:
Note: there will be no fallback for old '-rgb' suffix variables.

MikyungKim reviewed (commented):
Great work! It makes the code more consistent and simpler.

I found 2 additional places where `rgb()` is used. Would you please check again?
```
// colors.less
// Steps
// ---------------------------------------
@sand-steps-pageindicator-bg-color: rgb(111, 118, 128, 1);
```
```
//mixins.less
.sand-focus-ring() {
	transform: none;       
	outline: 12px solid rgb(230,230,230);
```

0x64 commented:
As the original goal of this PR is convert R,G,B format to #RGB, I did not check other rgb() calls. But, it looks better to clean up at this time so I updated the PR to convert them into a normal #RGB value without rgb().
And, I found rgba() calls that can be replaced with the standard rgb() from Firefox 52(2017), Chrome 65(2018), and Safari 12.1(2019). We will update the companion enact library to use Chrome 119+, Firefox 128+, and Safari 16.4 so I also replaced rgba() calls also.

0x64 commented:
Note: reverted rgba() calls for screenshot tests.

0x64 commented:
Updated: discussing @MikyungKim reverted rgba() to rgb() changes as it is not a scope of this PR and rgba() is also valid as a spec.

MikyungKim reviewed (approved):
LGTM

juwonjeong reviewed (approved):
LGTM
